### PR TITLE
Editorial: Refactor String.raw in anticipation of generalization

### PR DESCRIPTION
**Pull request by gibson042**
cf. https://github.com/tc39/proposal-string-cooked/pull/11#discussion_r879488509 and https://github.com/tc39/proposal-string-cooked/pull/11#discussion_r880732986

May be reviewed holistically or by commit.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -33480,23 +33480,22 @@ <h1>String.prototype</h1>
...
         <p>The `String.raw` function may be called with a variable number of arguments. The first argument is _template_ and the remainder of the arguments form the List _substitutions_. The following steps are taken:</p>
         <emu-alg>
-          1. Let _numberOfSubstitutions_ be the number of elements in _substitutions_.
+          1. Let _substitutionCount_ be the number of elements in _substitutions_.
           1. Let _cooked_ be ? ToObject(_template_).
-          1. Let _raw_ be ? ToObject(? Get(_cooked_, *"raw"*)).
-          1. Let _literalSegments_ be ? LengthOfArrayLike(_raw_).
-          1. If _literalSegments_ &le; 0, return the empty String.
-          1. Let _stringElements_ be a new empty List.
+          1. Let _literals_ be ? ToObject(? Get(_cooked_, *"raw"*)).
```
Comment (jridgewell) [inline spec.html:33485]: Nit: I'm calling these "blocks" in the `String.dedent` proposal. Essentially, a template is 1+ blocks, and each block can have multiple lines.

**Review by jridgewell — APPROVED** [review]

File: spec.html (hunk 1 above)
Comment (gibson042) [reply spec.html:33485]: Yeah, there doesn't seem to be consistency here. [String.raw](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.raw) refers to them as "literal segments", the [Template Literals grammar](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-template-literals) section arguably as "spans", [GetTemplateObject](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-gettemplateobject) and [TemplateStrings](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-static-semantics-templatestrings) and MDN as "[template] strings", etc.

Comment (jridgewell) [reply spec.html:33485]: I'd be fine with calling the segments or spans. But "literal lines" (`String.dedent` is concerned with the individual lines) sounds very confusing.

**Review by michaelficarra — APPROVED** [review]
LGTM

**Review by syg — APPROVED** [review]
lgtm

**Pull request merged**